healthcheck/handler.go

```diff
@@ -0,0 +1,50 @@
+package healthcheck
+
+import (
+	"net/http"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/gorilla/mux"
+	"sync"
+	"context"
+)
+
+const (
+	gracefulShutdownMsg = "graceful shutdown of healthcheck endpoint complete"
+)
+
+var httpServer *server.Server
+var serverErrors chan error
+var once sync.Once
+
+// NewHandler create and run the healthcheck API endpoint.
+func NewHandler(bindAddr string, errorChan chan error) {
+	once.Do(func() {
+		serverErrors = errorChan
```

Comment:
Consider just using errorChan and removing the serverErrors variable.

Reply:
I assigned it to a package var as I plan to use it from within the handle func if we get any errors - checking the db, checking kafka etc anything that might want to push an error back up to main.